cmd/horcrux/cmd/metrics.go

```diff
@@ -0,0 +1,83 @@
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"os"
+	"time"
+
+	"github.com/armon/go-metrics"
+	gmprometheus "github.com/armon/go-metrics/prometheus"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	tmlog "github.com/tendermint/tendermint/libs/log"
+)
+
+func AddPrometheusMetrics(mux *http.ServeMux) {
+	logger := tmlog.NewTMLogger(tmlog.NewSyncWriter(os.Stdout)).With("module", "metrics")
+
+	// Add raft metrics to prometheus
+	enableRaftMetrics := true
+	if enableRaftMetrics {
+		// PrometheusSink config w/ definitions for each metric type
+		cfg := gmprometheus.DefaultPrometheusOpts
+		sink, err := gmprometheus.NewPrometheusSinkFrom(cfg)
+		if err != nil {
+			logger.Error("Could not configure Raft Metrics")
+		}
+		defer prometheus.Unregister(sink)
+		_, err = metrics.NewGlobal(metrics.DefaultConfig("horcrux"), sink)
+		if err != nil {
+			logger.Error("Could not add Raft Metrics")
+		}
+	}
+
+	mux.Handle("/metrics", promhttp.Handler())
+	logger.Info("Prometheus Metrics Listening", "address", config.Config.DebugListenAddress, "path", "/metrics")
+}
+
+// EnableDebugAndMetrics - Initialization errors are not fatal, only logged
+func EnableDebugAndMetrics() {
+	logger := tmlog.NewTMLogger(tmlog.NewSyncWriter(os.Stdout)).With("module", "debugserver")
+
+	// Configure Shared Debug HTTP Server for pprof and prometheus
+	if len(config.Config.DebugListenAddress) == 0 {
+		logger.Error("debug-listen-address not defined")
+		return
```

Comment:
UX nit: Logging an error indicates to the operator they did something wrong. But if they don't set the listen address, then that was purposeful, not an error. How about making this an `Info` instead?

We probably want to expand the log message "debug-listen-address not defined; debug server disabled"

Reply:
Info with suggested message